### ISSUE TITLE
Can we improve the data binding error message

### ISSUE THREAD

**Issue by shafreenAnfar** [ballerina-platform/ballerina-standard-library#1864]
Right now the returned error message looks like this.

```
data binding failed: error("{ballerina/lang.typedesc}ConversionError",message="'map<json>' value cannot be converted to 'Student'")
```
In this case I am data binding to the below type,

```ballerina
type Student record {
    string Name;
    int Grade;
};
```

but in the request I only send the `Name` as below,
```
curl http://localhost:9090/hello/bindStudent -d '{ "Name": "John"}' -H "Content-Type:application/json"
```
Since the the data binding is failing because of the missing field `Grade `. Can't we make the error message a bit more specific and return that information ?

**Comment by shafreenAnfar** [comment]
Related issue https://github.com/ballerina-platform/ballerina-lang/issues/14810

**Comment by Nadeeshan96** [comment]
This also may gets fixed by ballerina-platform/ballerina-lang#32761.

**Comment by chamil321** [comment]
> This also may gets fixed by [ballerina-platform/ballerina-lang#32761](https://github.com/ballerina-platform/ballerina-lang/pull/32761).

@shafreenAnfar Can we close the issue?

**Comment by shafreenAnfar** [comment]
@chamil321 if it is fixed, let's close it.
